### PR TITLE
Add reference page for `mc du`

### PR DESCRIPTION
**Pull request by feorlen**
Create a new reference page for `mc du` and link it in the list of `mc` commands.

Staged:
http://192.241.195.202:9000/staging/DOCS-764/linux/html/reference/minio-mc/mc-du.html

Includes drive-by fixes to make the _Optional_ notations appear in a more consistent manner. (Including several otherwise unrelated pages.) There's more opportunistic fixes here than I might normally do in the same PR, but seemed better to get it over with. 

Fixes https://github.com/minio/docs/issues/764

### CODE REVIEW

File: source/includes/common-minio-mc.rst [hunk 1]
```diff
@@ -30,7 +30,7 @@ This command supports only global flags
 .. start-minio-mc-s3-compatibility
 
 The :program:`mc` commandline tool is built for compatibility with the AWS S3
-API and is tested MinIO and AWS S3 for expected functionality and behavior.
+API and is tested with MinIO and AWS S3 for expected functionality and behavior.
```
Comment (feorlen) [inline source/includes/common-minio-mc.rst:33]: Drive-by typo fix

File: source/includes/facts-versioning.rst [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .. start-rewind-desc
 
-*Optional* Directs |command| to operate only on the object version(s) that
+Directs |command| to operate only on the object version(s) that
```
Comment (feorlen) [inline source/includes/facts-versioning.rst:3]: Remove the manually formatted `*Optional*` from rewind and versions so it can be done with `:optional:` in the place this content is included. 

Other includes have the same issue, but these two are the ones my new page uses.

File: source/reference/minio-mc/mc-ls.rst [hunk 3]
```diff
@@ -42,7 +42,7 @@ results as the ``ls`` command.
 
          mc ls --recursive --versions myminio/mydata
 
-      The output resembles the following::
+      The output resembles the following:
```
Comment (feorlen) [inline source/reference/minio-mc/mc-ls.rst:45]: Another opportunistic typo fix.

File: source/reference/minio-mc/mc-du.rst [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+   Encrypt or decrypt objects using server-side encryption with client-specified keys. Specify key-value pairs as ``KEY=VALUE``.
+
+   - Each ``KEY`` represents a bucket or object.
+   - Each ``VALUE`` represents the data key to use for encrypting
+      object(s).
+
+   Enclose the entire list of key-value pairs passed to :mc-cmd:`~mc cat --encrypt-key` in double quotes ``"``.
+
+   :mc-cmd:`~mc cat --encrypt-key` can use the ``MC_ENCRYPT_KEY`` environment variable for retrieving a list of encryption key-value pairs as an alternative to specifying them on the command line.
```
Comment (ravindk89) [inline source/reference/minio-mc/mc-du.rst:109]: copypasta? mc cat -> mc du

**Review by ravindk89 — APPROVED** [review]
LGTM mod nit!

File: source/reference/minio-mc/mc-du.rst (hunk 4 above)
Comment (feorlen) [reply source/reference/minio-mc/mc-du.rst:109]: Well now you know which file I copied 😄

**Review by ravindk89 — APPROVED** [review]
Still LGTM with one more nit :D

File: source/reference/minio-mc/mc-du.rst [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+   Print the total for all folders N or fewer levels below the path specified in the command. Default is 0, for the specified path only.
+
+.. mc-cmd:: --encrypt-key
+   :optional:
+
+   Encrypt or decrypt objects using server-side encryption with client-specified keys. Specify key-value pairs as ``KEY=VALUE``.
+
+   - Each ``KEY`` represents a bucket or object.
+   - Each ``VALUE`` represents the data key to use for encrypting
+      object(s).
```
[the remark this answers is not in the document]
Comment (feorlen) [reply source/reference/minio-mc/mc-du.rst:105]: Good catch! I was reformatting these to one line, missed one.

**Comment by harshavardhana** [comment]
We should warn as the amount of data gets big, `mc du` might take a long time to finish @feorlen

File: source/reference/minio-mc/mc-du.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+   :class: copyable
+
+   mc du --rewind DURATION ALIAS/PATH
+
+- Replace :mc-cmd:`DURATION <mc du --rewind>` with the desired point-in-time in the past. For example, specify ``30d`` to show the disk usage 30 days prior to the current date.
+
+- Replace :mc-cmd:`ALIAS <mc du ALIAS>` with the 
+  :mc:`alias <mc alias>` of the S3-compatible host.
+
+- Replace :mc-cmd:`PATH <mc du ALIAS>` with the path to the bucket or folder on the S3-compatible host.
```
Comment (djwfyi) [inline source/reference/minio-mc/mc-du.rst:173]: Same commend as line 156.

File: source/reference/minio-mc/mc-du.rst [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use :mc-cmd:`mc du --recursive` to print a summary for each folder recursively:
+
+.. code-block:: shell
+   :class: copyable
+
+   mc du --recursive ALIAS/PATH
+
+- Replace :mc-cmd:`ALIAS <mc du ALIAS>` with the :mc:`alias <mc alias>` of the S3-compatible host.
```
Comment (djwfyi) [inline source/reference/minio-mc/mc-du.rst:189]: Same comment on this line and line 191 as in the above examples.

**Review by djwfyi — APPROVED** [review]
A few suggestions to remove some links so it's not a sea of blue fields when reading.
One additional drive-by typo I noticed that could be fixed with this PR.

Otherwise, LGTM! Great work on this one!
